binary-search-tree/node-js/inventory-management.js

```diff
@@ -18,16 +18,37 @@ class BinarySearchTree {
         this.root = null;
     }
 
-    search(id) {
-        console.error("you must implement search");
+    search(id, currentNode) {
+        if (id === null) {
+            return null;
+        } else {
+            return this.searchRecursive(id, currentNode)
```

Comment:
Now that you change the signature of the `search` method, which you shouldn't need to, imagine that this method is called by something that only knows the original signature e.g.

```javascript
bst.search(9)
```

Now inside the function `currentNode` will be set to `undefined`. That would lead to all sorts of issues, including the fact that we're no longer starting with `currentNode` as the root node.